### DISCUSSION TITLE
Add timestamp on status page

### DISCUSSION THREAD

**Discussion by francklap** [osm-search/Nominatim#2368] [Ideas]
Hello,

It would be great for monitoring if we could have timestamp on status page in json format. That way it is easier to create an alert when the update is too old. For now, at each update, we need to modify Nominatim.X.Y.Z/lib/nominatim/lib-php/website/status.php and add the line 'epoch' => (int)$epoch

if ($sOutputFormat == 'json') {
    header('content-type: application/json; charset=UTF-8');
}

try {
    $oStatus = new Nominatim\Status($oDB);
    $oStatus->status();

    if ($sOutputFormat == 'json') {
        $epoch = $oStatus->dataDate();
        $aResponse = array(
                      'status' => 0,
                      'message' => 'OK',
                      'data_updated' => (new DateTime('@'.$epoch))->format(DateTime::RFC3339),
                      'software_version' => CONST_NominatimVersion,
                      **'epoch' => (int)$epoch**
                     );

Thanks.

**Comment by lonvia** [comment]
I'm not aware that JSON defines a time format. You probably mean that you need the timestamp in epoch format. It's not feasible to return the date in all the possible different formats here that clients might need. You really should be converting the timestamp on the client side. Pretty much every programming language has support for parsing dates to do that.
